capstone/src/main/webapp/script.js

```diff
@@ -31,14 +32,61 @@ async function submitInput() {
   if (!langElement) {
     return;
   }
+
+  // Get the selected attributes
   const attributes = document.getElementById("available-attributes").getElementsByTagName('input');
   const requestedAttributes = []
   for (let attribute of attributes) {
     if (attribute.checked == true) {
       requestedAttributes.push(attribute.value);	
     }	
   }
-  await callPerspective(textElement.value, langElement.value, requestedAttributes);
+
+  // Make Perspective call for the entire submission and load data
+  const toxicityData = await callPerspective(textElement.value, langElement.value, requestedAttributes);
+  loadChartsApi(toxicityData);
+
+  // Get detailed analysis if requested
+  document.getElementById('analysis-container').innerHTML = '';
+  const radios = document.getElementsByName('analysisRadios');
+  var delimiter = null;
+  for (i = 0; i < radios.length; i++) {
+    if (radios[i].checked == true) {
+      delimiter = radios[i].value;
+      break;
+    }
+  }
+  if (delimiter != "") {
+    getAnalysis(textElement.value, langElement.value, delimiter); 
+  }
+}
+
+/** Prints detailed analysis of text by word or sentence */
+async function getAnalysis(text, lang, delimiter) {
+  const pieces = text.split(delimiter);
+  const result = createAnyElement('p', '');
+  for (i = 0; i < pieces.length; i++) {
+    if (pieces[i] != '') {
+      const response = await callPerspective(pieces[i], lang, ['TOXICITY']);
+      const score = response.attributeScores.TOXICITY.summaryScore.value;
```

Comment:
I think the reason why you sometimes get a nullPonter error is because `response` is a promise that may or may not have been evaluated at the time this line is executed. Maybe try using `await` in front of `response.att...` (although I'm not entirely sure about if it's syntactically correct).

Reply:
I think this may work while deployed, but in the testing environment, putting an `await` there doesn't make a difference in terms of the nullPointer exceptions. We should try to deploy and see what happens